### DISCUSSION TITLE
DAFoam recompiling

### DISCUSSION THREAD

**Discussion by hosseinnegahban** [mdolab/dafoam#312] [General]
Hi,

If we change anything in dafoam source code like objective function, is it mandatory to recompile DAFoam? If yes, should we recompile the whole package or only DAFoam.
I modified the objective function and recompiled DAFoam, but nothing changed.

I would appreciate if you help me in this regard.

Ragrads,
Hossein Negahban

**Comment by hosseinnegahban** [comment]
This is what I get when running my case. It does not recognize my new objective function. I have done exactly the same as DAFoam FAQ

"
Unknown DAObjFunc type clcd

Valid DAObjFunc types:

8
(
centerOfPressure
fieldInversion
force
meshQualityKS
moment
totalPressure
variableVolSum
wallHeatFlux
)

"

**Comment by friedenhe** [comment]
Please list all the changes you made to the code and all the commands you used to compile the new codes
